### PR TITLE
upgrade(moment/luxon): some work to move to luxon

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Enhancement**

This PR resolves []

### Description

This came up in the mobile practice, and we worked on it during two sessions. Thanks to everyone who helped (I might be misremembering but I'll try to tag people, please add yourself or others if I forgot 🙇.) @anandaroop @mdole @MounirDhahri @admbtlr @lordkiz @gkartalis @brainbicycle 

We can merge this, and then use this as an example to keep going.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a CHANGELOG.yml entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Some luxon love - pavlos, roop, george, brian

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Some luxon love - pavlos, roop, george, brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a9a707d14c130fd0f19c34162a85ea0ef8a9cbbc
</p>

File: src/lib/Scenes/MyBids/Components/SaleInfo.tsx [hunk 1]
```diff
@@ -17,40 +18,39 @@ export const SaleInfo = ({
...
     } else {
       return `${message} ${formatDate(endMoment)} at ${formatTime(endMoment)}`
     }
   }
 
-  const line2Message = (message: string, deadline: Moment): string | undefined => {
-    const hoursTillDeadline = deadline.diff(now, "hours")
-    if (now.isBefore(deadline) && hoursTillDeadline <= 10) {
+  const line2Message = (message: string, deadline: DateTime): string | undefined => {
+    const hoursTillDeadline = Math.floor(deadline.diff(now, "hours").as("hours"))
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyBids/Components/SaleInfo.tsx:50]: **question:** why was the `Math.floor` needed here?

**Review by MounirDhahri — APPROVED** [review]
Nice one 👏 Thanks!

File: src/lib/Scenes/MyBids/Components/SaleInfo.tsx (hunk 1 above)
Comment (pvinis) [reply src/lib/Scenes/MyBids/Components/SaleInfo.tsx:50]: it is.. luxon here behaves differently, and it returns something like `4.7488` hours. so we need to floor to get the hours right. 🤷

Comment (MounirDhahri) [reply src/lib/Scenes/MyBids/Components/SaleInfo.tsx:50]: perfect. thanks for explaining

**Pull request merged**